### ISSUE TITLE
CMSIS Documentation: Explain the mutex behaviour (recursive vs normal)

### ISSUE THREAD

**Issue by ilg-ul** [ARM-software/CMSIS#52]
The current [docs](http://www.keil.com/pack/doc/CMSIS/RTOS/html/group___c_m_s_i_s___r_t_o_s___mutex_mgmt.html) do not explicitly specify if the mutex is normal or recursive. However, RTX implements recursive mutexes (only), and the validator expect the mutex to  be recursive.

Suggestion:

* update the documentation to explain the expected mutex behaviour.

**Comment by ReinhardKeil** [comment]
almost identical with #53

**Comment by ilg-ul** [comment]
not really, issues prefixed with 'CMSIS Documentation' are intended to improve the documentation, which is very sparse in some areas.

#53 is a functional issue, intended to align the mutex behaviour with POSIX mutes specs.

**Comment by KeilChris** [comment]
Adding done flag here. Will be available in the CMSIS_5 repo within today.
